ambry-store/src/main/java/com.github.ambry.store/PersistentIndex.java

```diff
@@ -912,6 +912,27 @@ private long getAbsolutePositionInLogForOffset(Offset offset,
   }
 
   /**
+   * Fetches {@link LogSegment} names whose entries don't over lap with {@link Journal}. Returns {@code null}
+   * if there aren't any
+   * @return a {@link List<String>} of {@link LogSegment} names whose entries don't overlap with {@link Journal}.
+   * {@code null} if there aren't any
+   */
+  List<String> getLogSegmentsNotInJournal() {
+    LogSegment logSegment = log.getFirstSegment();
+    Offset firstOffsetInJournal = journal.getFirstOffset();
+    List<String> logSegmentNamesToReturn = new ArrayList<>();
+    while (logSegment != null) {
+      if (!logSegment.getName().equals(firstOffsetInJournal.getName())) {
+        logSegmentNamesToReturn.add(logSegment.getName());
+      } else {
+        break;
+      }
+      logSegment = log.getNextSegment(logSegment);
+    }
+    return logSegmentNamesToReturn.size() > 0 ? logSegmentNamesToReturn : null;
```

Comment:
No, no strong arguments for or against. I guess this is okay, since you would have to test if size > 0 anyways to determine whether to create `CompactionDetails`